### PR TITLE
Add 'get started' button to use-cases pages

### PR DESCRIPTION
**Pull request by deliaconstantino**
-  Closes gravitational/web#2755

## Fixes
Add "Get Started" button to header section of each `/use-cases/...` page by rendering `<Button />` from `<SectionHeader />` component

**Preview Pages at:**
- [`/use-cases/finance/`](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/finance/)
- [`/use-cases/saas/`](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/saas/)
- [`/use-cases/ecommerce-entertainment/`](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/ecommerce-entertainment/)
- [`/use-cases/aws/`](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/aws/)

- Update width of `<Button />` rendered by `<SectionHeader />` to match our global styling for mobile button width. This updates the button rendered in the `AccessPlane` section of each `case-study` page to render with 100% width on mobile
- **Preview [here](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/case-study/vmware-testimonial/)**
Screen Shot 2022-02-14 at 6 04 39 PM

### CODE REVIEW

**Comment by deliaconstantino** [comment]
> Let's modify slightly so that if there is a News Item, as in the case of the AWS page, that it is inline with the button rather than following it; that way we create less white space beneath the image (which can't resize much further...). 

For context, the page currently looks like this:
Screen Shot 2022-02-16 at 1 15 27 PM

The component tree looks like the following (with a number of props passed in):
```
<Section>
  <SectionHeader />
  <NewsItem />
  <AnnouncementBar />
</Section>
```

Right now the `<NewsItem />` ("Teleport is available in the -aws marketplace svg-") is rendered independently of the <`SectionHeader />` component. To implement the requested change, it looks like `<NewItem />` will need to be added as a prop to `<SectionHeader />`. 

Wanted to verify that we want to go down this route before adding further complexity to `<SectionHeader />`? Alternately `<SectionHeader />` could likely be simplified in a similar manner to the `<ProductBanner />` component that @alexwolfe and I were pairing on yesterday, with the bulk of the details assembled on the index.mdx page, rather than in components.

**Comment by deliaconstantino** [comment]
> modify slightly so that if there is a News Item, as in the case of the AWS page, that it is inline with the button rather than following it; that way we create less white space beneath the image (which can't resize much further...)

- create optional `inlineChild` prop for `<SectionHeader />` to pass in a child component. This `inlineChild` component is rendered inline with `<Button />` on full screen and slides below on mobile. Preview [here](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/aws/)

- Modified `<NewsItem />`(the component passed to `inlineChild`):
  - Replaced `<Centrator />` with `<Flex />`. `<Centrator />` uses `px="80px"` on full screen, which was causing layout issues when rendering the `<NewsItem />` as an inline element. `<NewsItem />` is currently only used on this page.

- Verified that all pages using SectionHeader render properly

**Review by C-STYR — CHANGES_REQUESTED** [review]
This was a difficult PR to review.  The major component `SectionHeader` is becoming more bloated as it is being asked to accommodate additional complexity.  It is probably in need of a major refactor/split. Within those constraints you did well to achieve the goal of adding a button, a seemingly simple task, yet not so easy!  

However this PR needs improvement.  Unnecessary props, props that aren't defined in the parent component, unused and duplicate imports are all present here.  Future PRs should undergo a line-by-line cleanup before submission. 

Additionally, the PR was submitted without in-code comments which would increase developer understanding and reduce code review time.  As an example, the introduced complexity of the function of the `inlineChild` prop and `ButtonSection` component should probably be explained in the code base.

File: pages/use-cases/aws/index.mdx [hunk 1]
```diff
@@ -26,24 +27,33 @@ import { SaaS } from "components/AnnouncementBar/data";
...
 import largorgs from "./assets/for-largorgs.svg";
+import awsMarketplaceLogo from "./assets/aws-marketplace-logo.svg"
 
 <Section bg="flatGray">
   <SectionHeader
     bg="transparent"
     title={meta.title}
     subtitle={meta.subtitle}
     description={meta.headerDescription}
+    isAWS={true}
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply pages/use-cases/aws/index.mdx:38]: removed, thank you

File: pages/use-cases/aws/index.mdx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import CenterCard from "components/CenterCard";
 import EasyStart from "components/EasyStart";
 import Feedback from "components/Feedback";
 import Image from "components/Image";
+import NextImage from "next/image";
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply pages/use-cases/aws/index.mdx:17]: removed

File: pages/use-cases/finance/index.mdx [hunk 3]
```diff
@@ -24,6 +24,7 @@ import { Finance } from "components/AnnouncementBar/data";
 import blockchain from "./assets/for-blockchain.svg";
 import fintech from "./assets/for-fintech.svg";
 import finserv from "./assets/for-finserv.svg";
+import Button from "components/Button";
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply pages/use-cases/finance/index.mdx:28]: removed thank you

File: components/NewsItem/NewsItem.tsx [hunk 4]
```diff
@@ -12,12 +12,15 @@ export interface NewsItemProps {
 }
 export const NewsItem = ({ newsImg = false, children }: NewsItemProps) => {
   return (
-    <Centrator flexDirection="row" mt={2} justifyContent="flex-start">
+    <Flex
+      flexDirection="row"
+      mt={2}
+      justifyContent="flex-start"
+      maxWidth="1200px"
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply components/NewsItem/NewsItem.tsx]: Reworked components to account for these use cases:

Changes are best seen with the vercel preview from [this deployment](https://teleport-docs-next-39c55f48h-gravitational.vercel.app/use-cases/aws/) as you can scroll across screen sizes. Added example usage for all three (current) version of `<NewsItem />`

- refactored `<Centrator />` to accept an optional `px` prop which allows the `<NewsItem />` rendered in `<SectionHeader />` to display without the default 80px of `px` - default `px` remain the same for all other use cases.
- refactored `<NewsItem />` to wrap and center content as screen size changes. The original had a bit of weird behavior at small screen sizes:
Screen Shot 2022-02-18 at 3 16 05 PM

- refactored `<SectionHeader />`:
  - renamed `inlineChild` prop to `newsItem` to better describe what it renders
  - moved `newsItem` and `<Button />` rendering back into `<SectionHeader />` component to follow existing pattern - let me know if comments are still needed here.
  - added `minWidth` to `newsItem` in `<SectionHeader />` to override the new wrapping from `<NewsItem />` which caused strange spacing in this use case

File: components/NewsItem/NewsItem.tsx [hunk 5]
```diff
@@ -12,12 +12,15 @@ export interface NewsItemProps {
 }
 export const NewsItem = ({ newsImg = false, children }: NewsItemProps) => {
   return (
-    <Centrator flexDirection="row" mt={2} justifyContent="flex-start">
+    <Flex
+      flexDirection="row"
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply components/NewsItem/NewsItem.tsx:24]: This comment applied to the previous iteration with `<Flex/ >`. It is now back to the original `<Centrator />` composition.

File: components/NewsItem/NewsItem.tsx [hunk 6]
```diff
@@ -12,12 +12,15 @@ export interface NewsItemProps {
...
+      mt={2}
+      justifyContent="flex-start"
```
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply components/NewsItem/NewsItem.tsx:27]: Same as above.

File: components/NewsItem/NewsItem.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (deliaconstantino) [reply components/NewsItem/NewsItem.tsx]: Edited one final thing: `paddingTop` on the "awsMarketplace" svg link to center it vertically with the text that it follows. Preview `use-cases/aws` page [here](https://teleport-docs-next-git-delia-add-get-start-a5f16c-gravitational.vercel.app/use-cases/aws/)

**Review by C-STYR — APPROVED** [review]

**Pull request merged**